pmlc/dialect/pxa/transforms/tile.cc

```diff
@@ -2,12 +2,29 @@
 
 #include <iostream>
 
+#include "mlir/include/mlir/IR/BlockAndValueMapping.h"
 #include "pmlc/dialect/pxa/transforms/tile.h"
 
 namespace pmlc::dialect::pxa {
 
 using mlir::AffineParallelOp;
 
+void replaceAllUses(Operation* op, mlir::BlockAndValueMapping& mapper) {
```

Comment:
So my whole point of not calling clone was to avoid recursion.  Basically, what I'm say is: For a given value (such as the outer index variable) you can in O(uses) time walk over all the uses of the value, and fix them in place, so then it's not depending on how many ops are in the inner body, but just how many times the index is used (which will typically be much smaller).  Also, since you are just updating the value in place, no allocation is done at all.  The entire point of MLIR's complex use-def tracking system is largely to avoid walking over ops at all.

Reply:
ok, I'll try.